File: source/repos/BlackJack/BlackJack.ViewModels/HistoryViews/GetBotStepsHistoryView.cs
using BlackJack.DataAccess.Enums;
using System;
using System.Collections.Generic;

namespace BlackJack.ViewModels.HistoryViews
{
    public class GetBotStepsHistoryView
    {
        public Guid GameId { get; set; }
        public List<BotGetBotStepsHistoryViewItem> Bots { get; set; }

        public GetBotStepsHistoryView()
        {
            Bots= new List<BotGetBotStepsHistoryViewItem>();
        }
    }
    public class BotGetBotStepsHistoryViewItem
    {
        public string Name { get; set; }
        public List<CardBotStepsHistoryViewItem> Steps { get; set; }
    }
 
    public class CardBotStepsHistoryViewItem
    {
        public CardRankType Rank { get; set; }
        public CardSuitType Suit { get; set; }
    }
}

File: source/repos/BlackJack/BlackJack.WEB/ClientApp/src/app/shared/entities/get-bot-steps-history.view.ts
export class GetBotStepsHistoryView{
    gameId:string;
    bots:BotGetBotStepsHistoryViewItem[];
}
export class BotGetBotStepsHistoryViewItem{
    name:string;
    steps:CardGetBotStepsHistoryViewItem[];
}
export class CardGetBotStepsHistoryViewItem{
    rank:number;
    suit:number;
}